src/main/scala/Ctx.scala

```diff
@@ -1,45 +1,47 @@
 import akka.NotUsed
-import akka.util.Timeout
-import schema.MutationError
 import akka.actor.ActorRef
-import generic.Event
-import generic.MemoryEventStore._
-import generic.View.{Get, GetMany}
 import akka.pattern.ask
 import akka.stream.OverflowStrategy
 import akka.stream.scaladsl.Source
-import org.reactivestreams.Publisher
+import akka.util.Timeout
+import generic.Event
+import generic.PersistentEventStore._
+import generic.View.{Get, GetMany}
+import schema.MutationError
 
+import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
+import scala.language.postfixOps
 
 case class Ctx(
   authors: ActorRef,
   articles: ActorRef,
   eventStore: ActorRef,
-  eventStorePublisher: Publisher[Event],
+  events: Source[Event, NotUsed],
   ec: ExecutionContext,
   to: Timeout
 ) extends Mutation {
   implicit def executionContext = ec
   implicit def timeout = to
 
+  // TODO why buffer?
   lazy val eventStream: Source[Event, NotUsed] =
-    Source.fromPublisher(eventStorePublisher).buffer(100, OverflowStrategy.fail)
+    events.buffer(100, OverflowStrategy.fail).backpressureTimeout(3 seconds)
```

Comment:
Yeah, what is even more important is that this buffer isolates the error handling, I think. So if client disconnects or has an error, it does not take down the whole system with it.


Regarding `BroadcastBub`, Really happy that it's it. As I was implementing this example, nothing of this sorts was available. i even posted it on the mailing list:

https://groups.google.com/forum/#!topic/akka-user/DiSyunt5XMs

I guess now a big portion of client handing can be improved with new features in akka-streams